_lessons/python-scatter/06-profiling_with_MAP.md

```diff
@@ -1,77 +1,79 @@
 ---
 layout: post
-title: Profiling_MAP
+title: Profiling with MAP
 permalink: /python-scatter/profiling_MAP
 chapter: python-scatter
 ---
 
 ## Objectives
 
-Learn how to profile parallel codes using ARM MAP:
+You will:
 
-* learn how to use MAP
-* visualise the profiling data
-* interpret the profiling results
+* learn how to use MAP to profile an MPI code
+* learn how to interpret MAP profiling data
 
 ## Introduction to profiling parallel codes
 
 Parallel profiling tools provide information about how much time is spent in different parts of your code by different threads/processes.
-There are 2 major ways to collect profiling data: *sampling* and *tracing*. Sampling is a statistical method. During run time the so called *stack trace* is analysed regularly. In contast to that tracing measures actual run time of instrumented parts with a well defined start and end. The data from all threads/processes are gathered and analysed. Thus we can not only see which part of the code is most time consuming, but also how this performs on different processes.
+There are two major ways to collect profiling data: 
+ * *sampling*: data are statistically harvested during run time to determine which parts of the code are most time consuming. This is useful for identiying the hotspots in a code (see [profiling](https://nesi.github.io/perf-training/python-scatter/profiling)).
+ * *tracing*: records the activity of instrumented parts of the code in time. This is useful to determine what each thread and process is doing at a particaular point in time. Tracing can reveal load balancing issues or the communication patterns of an MPI program. 
```

Comment:
thanks for visually highlighting.
Unfortunately it was not clear enough. Tracing measures run time, while sampling "just" statistically analyse call stack snapshots. 
Both have in common that they collect data from all processes ...

Reply:
Sorry, that additional comment was just for you. 
The point is, that both methods determine what each thread and processes are doing...

Reply:
discussed with Mandes and Wolfgang. Decided we would take out this section